clang/include/clang/Driver/Options.td

```diff
@@ -5721,6 +5721,14 @@ def sycl_std_EQ : Joined<["-"], "sycl-std=">, Group<sycl_Group>,
   MarshallingInfoEnum<LangOpts<"SYCLVersion">, "SYCL_None">,
   ShouldParseIf<!strconcat(fsycl_is_device.KeyPath, "||", fsycl_is_host.KeyPath)>;
 
+def fsycl_default_sub_group_size
+    : Separate<["-"], "fsycl-default-sub-group-size">,
+      HelpText<"Set the default sub group size for SYCL kernels">,
+      Flags<[CC1Option]>;
```

Comment:
Should this use the new marshalling stuff?

Reply:
I couldn't figure out how to get this to control 2 different flags based on its input.  Got any hints :) ?

Basically, this can be a number, 'primary' or 'automatic', which causes different values for 2 different lang-opts flags.